src/Action/GraphqlEntrypointAction.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Action;
+
+use ApiPlatform\Core\Bridge\Graphql\Executor;
+use ApiPlatform\Core\Bridge\Graphql\Type\SchemaBuilder;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * GraphQL API entrypoint.
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class GraphqlEntrypointAction
+{
+    private $requestStack;
+    private $schemaBuilder;
+    private $executor;
+
+    public function __construct(RequestStack $requestStack, SchemaBuilder $schemaBuilder, Executor $executor)
+    {
+        $this->requestStack = $requestStack;
+        $this->schemaBuilder = $schemaBuilder;
+        $this->executor = $executor;
+    }
+
+    public function __invoke(): JsonResponse
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        // @TODO Manage GET request.
+        $input = json_decode($request->getContent(), true);
```

Comment:
You should check that the incoming `Content-Type` is `application/json` or `application/graphql`:

* if it's `application/json`, the current behavior is OK
* if it's `application/graphql`, the raw content is a GraphQL query (so no `json_decode` needed)
* throw an error for all other content types (maybe can you also serve https://github.com/graphql/graphiql if the content type is `text/html`)?